benches/search_comparison.rs

```diff
@@ -165,36 +165,32 @@ fn search_bench_case<const MAX: usize, T, Coll>(
     setup_fun: impl Fn(Vec<T>) -> Coll,
     search_fun: impl Fn(&Coll, T) -> Option<&T>,
     group: &mut BenchmarkGroup<WallTime>,
-    i: &usize,
+    size: &usize,
     duplicates: bool,
 ) where
-    T: TryFrom<usize> + Ord + std::ops::Rem<Output = T> + num_traits::ops::wrapping::WrappingMul,
+    T: TryFrom<usize> + Ord + Clone,
     <T as TryFrom<usize>>::Error: core::fmt::Debug,
 {
-    group.bench_with_input(BenchmarkId::new(name, i), i, |b, i| {
-        let size = *i;
+    group.bench_with_input(BenchmarkId::new(name, size), size, |b, &size| {
+        // increasing sequence of even numbers, bounded by MAX
+        let iter = (0usize..)
+            .map(|i| std::cmp::min(i * 2, MAX))
+            .map(|i| T::try_from(i).unwrap());
+
         let v: Vec<T> = if duplicates {
-            (0..*i)
-                .map(|int| {
-                    let int = std::cmp::min(int, MAX);
-                    T::try_from(int % 256).unwrap()
-                })
+            iter
+                // Repeat each items 16 times
+                .flat_map(|i| std::iter::repeat(i).take(16))
+                .take(size)
                 .collect()
         } else {
-            (0..*i)
-                .map(|int| {
-                    // Generate only even numbers to provide a ~50% hit ratio in the benchmark
```

Comment:
I'd like to keep this comment to explain the `i * 2`.

Reply:
Kept the comment